parking_permits/views.py

```diff
@@ -322,6 +330,68 @@ def post(self, request, format=None):
         return Response({"message": "Order received"}, status=200)
 
 
+class SubscriptionView(APIView):
+    @swagger_auto_schema(
+        operation_description="Used as an webhook by Talpa in order to send a subscription notification.",
+        request_body=SubscriptionSerializer,
+        security=[],
+        responses={
+            200: openapi.Response(
+                "Subscription event received response", MessageResponseSerializer
+            )
+        },
+        tags=["Subscription"],
+    )
+    @transaction.atomic
+    def post(self, request, format=None):
+        logger.info(
+            f"Subscription event received. Data = {json.dumps(request.data, default=str)}"
+        )
+        talpa_order_id = request.data.get("orderId")
```

Comment:
where do you check if our request is valid, you should check that order id given on our side and fetch user id from our admin endpoint and check that userId matches same as in payload.

Reply:
Valid point. Indeed this separate validation should be added.
I'll add this.

Reply:
Validation added in [c37032c](https://github.com/City-of-Helsinki/parking-permits/pull/358/commits/c37032c91b181bcafe7432437cf6c039b99a54d2)